Mathlib/Dynamics/FixedPoints/Increasing.lean

```diff
@@ -0,0 +1,98 @@
+/-
+Copyright (c) 2024 Shuhao Song. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Shuhao Song
+-/
+import Mathlib.Data.Fintype.Card
+
+/-!
+# Fixed points of function `f` with `f(x) ≥ x`
+
+In this file we consider the fixed points of function `f : α → α` with `f(x) ≥ x`. If `α` is
+a finite type, then the sequence `x, f(x), f(f(x)), ...` will eventually be a constant.
+
+# Main definitions
+* `fixedPointIndex`: The smallest `n` such that the sequence `x, f(x), f(f(x)), ...`
+  becomes a constant after this index.
+* `eventuallyValue`: The value that the sequence converges to after sufficient number
+  of iterations of `f` on `x`.
+-/
+namespace Function
+
+variable {α : Type*} [PartialOrder α] [Fintype α] {f : α → α} (hf : ∀ x, x ≤ f x)
+
+/-- If `x` is a fixed point of `f`, then the nᵗʰ iteration of `f` on `x` is `x`. -/
+lemma iterate_fixed_point {x : α} (hx : IsFixedPt f x) (n : ℕ) : f^[n] x = x := by
+  induction n with
+  | zero => simp
+  | succ m ih =>
+    rw [add_comm, iterate_add_apply]
+    simpa [ih] using hx
+
+/-- If `f^[m + 1] x = f^[m] x`, then the sequence `n ↦ f^[n] x` is constant after index `m`. -/
+lemma iterate_fixed_point_from_index
+    {m n : ℕ} {x : α} (hx : f^[m + 1] x = f^[m] x) (hn : n ≥ m) : f^[n] x = f^[m] x := by
+  rw [add_comm, iterate_add_apply] at hx
+  rw [show n = n - m + m by omega, iterate_add_apply]
+  apply iterate_fixed_point hx
+
+/-- Monotonicity of `n ↦ f^[n] x`. -/
+lemma monotone_iterate (x : α) : Monotone (fun n => f^[n] x) := by
```

Comment:
We have [Monotone.monotone_iterate_of_le_map](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Iterate.html#Monotone.monotone_iterate_of_le_map)